packages/plugin-backgroundcolor/src/BackgroundColor.ts

```diff
@@ -0,0 +1,164 @@
+import { JWPlugin, JWPluginConfig } from '../../core/src/JWPlugin';
+import { CommandParams } from '../../core/src/Dispatcher';
+import { VNode } from '../../core/src/VNodes/VNode';
+import { Loadables } from '../../core/src/JWEditor';
+import { Keymap } from '../../plugin-keymap/src/Keymap';
+import { Inline } from '../../plugin-inline/src/Inline';
+import { InlineNode } from '../../plugin-inline/src/InlineNode';
+import { getStyles, setStyles, setStyle, getStyle, removeStyle } from '../../utils/src/utils';
+import { VRange } from '../../core/src/VRange';
+import { Format } from '../../plugin-inline/src/Format';
+
+export interface ColorParams extends CommandParams {
+    color?: string; // css-valid color name
+}
+
+interface BackgroundColorConfig extends JWPluginConfig {
+    defaultBackgroundColor?: string;
+}
+
+const defaultConfiguration = {
+    defaultBackgroundColor: 'white',
+};
+
+export class BackgroundColor<T extends BackgroundColorConfig = BackgroundColorConfig> extends JWPlugin<T> {
+    static hasColor(node: VNode): boolean;
+    static hasColor(backgroundColor: string, node: VNode): boolean;
+    static hasColor(backgroundColor: string | VNode, node?: VNode): boolean {
+        if (backgroundColor instanceof VNode) {
+            node = backgroundColor;
+        }
+        const nodeBackgroundColor = getStyle(node, 'background-color');
+        if (!nodeBackgroundColor || backgroundColor instanceof VNode) {
+            return !!nodeBackgroundColor;
+        } else {
+            return nodeBackgroundColor === backgroundColor;
+        }
+    }
+    static dependencies = [Inline];
+    configuration = { ...defaultConfiguration, ...this.configuration };
```

Comment:
That's rather out of scope of this PR but what about having something automatic like this:
```typescript
class JWPlugin<T extends JWPluginConfiguration> {
    defaults: T;
    configuration: T = { ...this.defaults, ...this.configuration};
}
```

Reply:
Why not indeed. Out of scope indeed :-) Let's put a pin on it.